odoo/addons/base/models/res_users.py

```diff
@@ -181,6 +181,10 @@ def write(self, vals):
             self.env['res.users'].has_group.clear_cache(self.env['res.users'])
         return super(Groups, self).write(vals)
 
+    def unlink(self):
+        if self.mapped('rule_groups'):
```

Comment:
`mapped` is now automatic with relational fields:
```suggestion
        if self.rule_groups:
```